zmq.hpp

```diff
@@ -365,7 +365,12 @@ class message_t
         int rc = zmq_msg_init_size(&msg, size_);
         if (rc != 0)
             throw error_t();
-        memcpy(data(), data_, size_);
+        if (size_)
+        {
+            // this constructor allows (nullptr, 0),
```

Comment:
Does the constructor allow that? Should it?

We can add the check to be on the extremely safe side anyway. (Does this actually cause UB on any existing implementation? I know the standard allows ot to be UB.)

Reply:
I think it is common for constructing an empty message (in generic code). There is a similar check in libzmq in `zmq_recv`.

It will optimize away any subsequent null checks on gcc, see https://godbolt.org/z/vyx6eG, but not on clang.